services/QuillLMS/app/services/classlink_integration/instructor_linker.rb

```diff
@@ -11,17 +11,25 @@ def initialize(persisted_user:, classlink_instance_id:, school_sourced_id:)
     end
 
     def run
-      return unless persisted_user.role == ::User::ADMIN
+      return unless [::User::ADMIN, ::User::TEACHER].include?(persisted_user.role)
       return unless school_sourced_id
+      return unless classlink_instance_school
 
-      cis = ClasslinkInstanceSchool.find_by(classlink_instance_id:, external_id: school_sourced_id)
-      return unless cis
+      persisted_user.role == ::User::ADMIN ? create_admin_linkage : create_teacher_linkage
+    end
 
-      SchoolsAdmins.find_or_create_by!(user_id: persisted_user.id, school_id: cis.school_id)
+    private def create_admin_linkage
+      SchoolsAdmins.find_or_create_by!(user_id: persisted_user.id)
+        .update!(school_id: classlink_instance_school.school_id)
       auto_verify_admin
     end
 
-    def auto_verify_admin
+    private def create_teacher_linkage
+      SchoolsUsers.find_or_initialize_by(user_id: persisted_user.id)
+        .update!(school_id: classlink_instance_school.school_id)
+    end
+
+    private def auto_verify_admin
       admin_info = AdminInfo.find_or_create_by!(user_id: persisted_user.id)
       return if admin_info.approval_status == ::AdminInfo::APPROVED
```

Comment:
Another law of demeter one from an older commit, but might as well fix it while you're in here:

`def approved? = approval_status == APPROVED`

then line 34 becomes: `return if admin_info.approved?`